src/steps/nagivation.steps.ts

```diff
@@ -48,6 +48,47 @@ Given(
   },
 );
 
+Given(
+  'I navigate to the Rope Detail Page for an In Service rope with Serial number {string}',
+  async function (this: IHoistWorld, serialNumber: string) {
+    const page = this.page!;
+    console.log(`Searching for In Service rope: ${serialNumber}`);
+
+    try {
+      await page
+        .getByRole('gridcell', { name: serialNumber, exact: true })
+        .click({ timeout: 5000 });
+    } catch {
+      const statusHeader = page.locator('.ag-cell-label-container').filter({ hasText: 'Status' });
+      await statusHeader.locator('.menu-icon').click();
+      try {
+        await page.getByLabel('(Select All)').uncheck();
+      } catch (error) {
```

Comment:
I have a question why is this so complicated?

Reply:
the Request Extension form is only enabled for In Service ropes so this ensures that we're only checking for ropes with 'In Service' status

this part just locates the `div` that has the heading 'Status' because every other table header also has a filter icon

```
const statusHeader = page.locator('.ag-cell-label-container').filter({ hasText: 'Status' });
await statusHeader.locator('.menu-icon').click();
```

    
<img width="1006" alt="Screenshot 2024-12-20 at 16 48 52" src="https://github.com/user-attachments/assets/eee4a43e-5806-48c0-b2b7-45af6e60484b" />

this try-catch block ignores the playwright error of unchecking a checkbox thats already unchecked (just in case it is because of a previous test. this happened when i was testing this locally so i included it) 

```
  try {
        await page.getByLabel('(Select All)').uncheck();
      } catch (error) {
        if (
          error instanceof Error
          && error.message.includes('Clicking the checkbox did not change its state')
        ) {
          // Ignore the "did not change its state" error
          return;
        }
        throw error;
      }
```